src/backend/base/langflow/components/tools/astradb_cql.py

```diff
@@ -150,14 +150,21 @@ def build_tool(self) -> StructuredTool:
             Tool: The built AstraDB tool.
         """
         schema_dict = self.create_args_schema()
-        return StructuredTool.from_function(
+        structured_tool = StructuredTool.from_function(
             name=self.tool_name,
             args_schema=schema_dict["ToolInput"],
             description=self.tool_description,
             func=self.run_model,
             return_direct=False,
         )
 
+        # Convert the StructuredTool to a regular Tool
+        return Tool(
+            name=structured_tool.name,
+            func=structured_tool.func,
+            description=structured_tool.description,
+        )
+
```

Comment:
No need to convert to tool.

Reply:
Hey @ogabrielluiz the report was that it couldn't be connected to the Agent tool input, so is there another bug here? Should StructuredTools be able to be connected to Agent tool input / other tool inputs?

Reply:
What matters is the type hint. We use StructuredTool in all components.

Reply:
Got it, my mistake there. i just fixed it to return the structured tool but type hint it as Tool. i kind of thought Ruff wouldn't like that to be honest but it seems solid!

Reply:
Me too! hahahah But it probably does not care because they are subclasses of BaseTool.